akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -472,12 +475,26 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed with Eventua
       "not exceed stream-level window while sending after SETTINGS_INITIAL_WINDOW_SIZE changed when window became negative through setting" in pending
 
       "eventually send WINDOW_UPDATE frames for received data" in pending
+
+      "multiple WINDOW_UPDATEs exceeding total demand above 2^31-1 MUST cause FLOW_CONTROL_ERROR (spec 6.9.1)" in new TestSetup with RequestResponseProbes {
+        override def safeUpdate(update: Int ⇒ Int): Int ⇒ Int = { oldValue ⇒ update(oldValue) }
+
+        sendHEADERS(1, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+
+        sendWINDOW_UPDATE(1, Math.pow(2, 30).toInt)
+        sendWINDOW_UPDATE(1, Math.pow(2, 30).toInt)
+        sendWINDOW_UPDATE(1, Math.pow(2, 30).toInt)
+
+        val (lastStreamId, errorCode) = expectGOAWAY()
```

Comment:
Also add a test for the connection-level one (which should then indeed be a GOAWAY).